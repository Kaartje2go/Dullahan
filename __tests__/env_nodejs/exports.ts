import {sync as globSync} from 'glob';

globSync('packages/*/').forEach((path: string) => {
    const packageName = `@k2g/${/\/(dullahan[a-z0-9-]*)\//u.exec(path)![1]}`;
    const packageExports = require(packageName) ?? ({
        empty: null
    });
    const packageEntries = Object.entries(packageExports);

    describe(packageName, () => {
        it.each(packageEntries)('%s should remain the same', (key, value) => {
            expect(value).toMatchSnapshot();
        });
    });
});
