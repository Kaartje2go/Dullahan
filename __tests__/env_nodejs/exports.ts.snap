// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@k2g/dullahan AdapterError should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan AssertionError should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan DullahanAdapter should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan DullahanAdapterDefaultOptions should remain the same 1`] = `
Object {
  "debug": true,
  "headless": false,
}
`;

exports[`@k2g/dullahan DullahanApi should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan DullahanError should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan DullahanErrorMessage should remain the same 1`] = `
Object {
  "ACTIVE_BROWSER": "Browser has already been opened!",
  "NO_BROWSER": "Browser has been closed, or has not yet been opened!",
  "NO_ELEMENT": [Function],
}
`;

exports[`@k2g/dullahan DullahanPlugin should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan DullahanRunner should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan DullahanTest should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan attempt should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan cli should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan displayPointer should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan emitFakeEvent should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan findElement should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan getBoundingClientRect should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan getElementAttributes should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan getElementProperties should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan getElementStyles should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan isDullahanConfig should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan normalizeConfigEntry should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan requireDependency should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan scrollToElement should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan sleep should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan waitForReadyState should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-adapter-playwright DullahanAdapterPlaywrightDefaultOptions should remain the same 1`] = `
Object {
  "browserName": "chromium",
  "debug": true,
  "headless": false,
}
`;

exports[`@k2g/dullahan-adapter-playwright default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-adapter-puppeteer DullahanAdapterPuppeteerDefaultOptions should remain the same 1`] = `
Object {
  "browserName": "chrome",
  "debug": true,
  "headless": false,
  "useTouch": false,
}
`;

exports[`@k2g/dullahan-adapter-puppeteer default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-adapter-selenium-3 DullahanAdapterSelenium3DefaultOptions should remain the same 1`] = `
Object {
  "browserName": "chrome",
  "debug": true,
  "headless": false,
}
`;

exports[`@k2g/dullahan-adapter-selenium-3 default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-adapter-selenium-4 DullahanAdapterSelenium4DefaultOptions should remain the same 1`] = `
Object {
  "browserName": "chrome",
  "debug": true,
  "headless": false,
  "useActions": true,
}
`;

exports[`@k2g/dullahan-adapter-selenium-4 default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-aws-s3 default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-browserstack default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-github default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-gitlab default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-report-html default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-report-markdown default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-slack default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-plugin-xvfb default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-runner-aws-lambda default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-runner-development default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-runner-parallel default should remain the same 1`] = `[Function]`;

exports[`@k2g/dullahan-runner-synchronous default should remain the same 1`] = `[Function]`;
